tui/installer/__init__.py

```diff
@@ -72,20 +75,20 @@ def preserve_settings(answers):
     def preserve_timezone(answers):
         if not_preserve_settings(answers):
             return False
-        if 'installation-to-overwrite' not in answers:
-            return False
-        settings = answers['installation-to-overwrite'].readSettings()
-        return 'timezone' in settings and 'request-timezone' not in settings
+        if 'installation-to-overwrite' in answers and answers['installation-to-overwrite'].settingsAvailable():
+            settings = answers['installation-to-overwrite'].readSettings()
+            return 'timezone' in settings and 'request-timezone' not in settings
+        return False
     not_preserve_timezone = lambda a: not preserve_timezone(a)
 
     def ha_enabled(answers):
         settings = {}
-        if 'installation-to-overwrite' in answers:
+        if 'installation-to-overwrite' in answers and answers['installation-to-overwrite'].settingsAvailable():
```

Comment:
ditto
```suggestion
        if (
            "installation-to-overwrite" in answers
            and answers["installation-to-overwrite"].settingsAvailable()
        ):
```

Reply:
I've opted for the `and \` convention in keeping with the rest of this code base
Same for other comments